<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>788</width>
    <height>609</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>TDT4255 Programming Utility</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>FPGA bitfile:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="txtBitfile">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>451</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btnSelBitfile">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>10</y>
      <width>41</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnUpload">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>10</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Upload</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="prgBitfileProgress">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>591</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>121</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Upload Progress:</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabExSel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>761</width>
      <height>501</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Exercise 0</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>401</width>
        <height>381</height>
       </rect>
      </property>
      <property name="title">
       <string>RPN Program</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <widget class="QLineEdit" name="txtRPNExpression">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>350</y>
         <width>211</width>
         <height>27</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="btnConvertInstrs">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>350</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Add instructions</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>330</y>
         <width>211</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>RPN expression (e.g 3 -4 + 1 ~):</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblInstrCount">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>181</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Instructions: (count = 0)</string>
       </property>
      </widget>
      <widget class="QListWidget" name="lstInstructions">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>361</width>
         <height>181</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="btnClrInstrs">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>270</y>
         <width>99</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="grpProcControl">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>70</y>
        <width>331</width>
        <height>381</height>
       </rect>
      </property>
      <property name="title">
       <string>Processor Control</string>
      </property>
      <widget class="QPushButton" name="btnReadStackTop">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>70</y>
         <width>71</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Read</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>81</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Stack Top:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="txtStackTop">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>70</y>
         <width>151</width>
         <height>27</height>
        </rect>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>160</y>
         <width>298</width>
         <height>128</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="btnWriteProgram">
          <property name="text">
           <string>Write current RPN program into memory</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnExecOne">
          <property name="text">
           <string>Execute single instruction</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnExecAll">
          <property name="text">
           <string>Execute all instructions</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QProgressBar" name="progressBar">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>340</y>
         <width>291</width>
         <height>23</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>310</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Write Program Progress:</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="lblConnectionStatusEx0">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>311</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Exercise framework status:  &lt;unknown&gt;</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnCheckConnEx0">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>10</y>
        <width>138</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Check</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Exercise 1 or 2</string>
     </attribute>
     <widget class="QPushButton" name="btnCheckConnEx1">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>10</y>
        <width>138</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Check</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblConnectionStatusEx1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>311</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Exercise framework status:  &lt;unknown&gt;</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="grpEx1DataMem">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>310</y>
        <width>741</width>
        <height>151</height>
       </rect>
      </property>
      <property name="title">
       <string>Data Memory</string>
      </property>
      <widget class="QHexEdit" name="dataMemDisplay">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>30</y>
         <width>401</width>
         <height>121</height>
        </rect>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="btnLoadDataFromFile">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>30</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Load from file</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnReadData">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>90</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Read from CPU</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnWriteData">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>60</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Write to CPU</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnSaveDataToFile">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>120</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Save to file</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelDataAddr">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>addr = 0</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelDataValSigned">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>signed = 0</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelDataValHex">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>hex = 0</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="grpEx1InstMem">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>741</width>
        <height>151</height>
       </rect>
      </property>
      <property name="title">
       <string>Instruction Memory</string>
      </property>
      <widget class="QHexEdit" name="instMemDisplay">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>30</y>
         <width>401</width>
         <height>121</height>
        </rect>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="btnSaveInstToFile">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>120</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Save to file</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnLoadInstFromFile">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>30</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Load from file</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnReadInst">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>90</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Read from CPU</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnWriteInst">
       <property name="geometry">
        <rect>
         <x>620</x>
         <y>60</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Write to CPU</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelInstAddr">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>addr = 0</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelInstValSigned">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>signed = 0</string>
       </property>
      </widget>
      <widget class="QLabel" name="lblSelInstValHex">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>171</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>hex = 0</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="grpEx1ProcCtrl">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>741</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>Processor Control and Status</string>
      </property>
      <widget class="QPushButton" name="btnEx1ProcStart">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>30</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Run Processor</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnEx1ProcReset">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>30</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnEx1ProcStop">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>30</y>
         <width>121</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Stop Processor</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Actions:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>151</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Read/Write Memory:</string>
       </property>
      </widget>
      <widget class="QProgressBar" name="prgReadWriteBuf">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>70</y>
         <width>551</width>
         <height>23</height>
        </rect>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QHexEdit</class>
   <extends>QScrollArea</extends>
   <header>qhexedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
